XX.md

```diff
@@ -81,34 +94,135 @@ Dedicated event for shipping details:
+    ["price", "0", "USD"],
+    ["zone", "US"],
+    ["region", "US-FL"],
+    ["service", "pickup"],
+    ["location", "456 Ocean Drive, Miami Beach, FL 33139"],
+    ["g", "dhwv1zp8k"],
+  ]
+}
+```
+
+Standard shipping option:
+```jsonc
+{
+  "kind": 30406,
+  "created_at": 1703187600,
+  "content": "Standard domestic shipping within Florida",
+  "tags": [
+    ["d", "fl-standard"],
+    ["name", "Florida Standard Shipping"],
+    ["price", "5.99", "USD"],
+    ["zone", "US"],
+    ["region", "US-FL"],
+    ["service", "standard"],
+    ["duration", "24", "72"],
+    ["weight-max", "30", "kg"],
+    ["dim-max", "120", "60", "60", "cm"],
+    ["price-weight", "0.75", "USD", "kg"]
+  ]
+}
+```
+
+#### Shipping Notes
+
+1. For merchants with multiple pickup locations:
+   - Create separate shipping option events for each physical location
+   - Each location should have its own unique `d` tag identifier
+   - Product listings can reference multiple pickup options
+
+2. Clients should:
+   - Group pickup locations by merchant when displaying options
+   - Use geohash data to show pickup locations on a map
+   - Sort pickup locations by distance from user when possible
+
+3. Location identification:
+   - Each pickup location must have both `location` and `g` tags
+   - The `location` tag contains human-readable address
+   - The `g` tag contains geohash for precise positioning
+
 ## Order Communication Flow
 
-- Order processing uses [NIP-17](17.md) encrypted direct messages. Message direction is determined by the `p` tag - when sent from buyer to merchant, `p` contains the merchant's pubkey, and when sent from merchant to buyer, `p` contains the buyer's pubkey. 
+- Order processing and communication uses [NIP-17](17.md) encrypted direct messages.
+  - Kind `14` is used for regular communication, enabling users to maintain a conversation. The subject can be an order ID or left blank, depending on the context.
+  - Kind `15` is used for order processing and business logic.
+- Message direction is determined by the `p` tag - when sent from buyer to merchant, `p` contains the merchant's pubkey, and when sent from merchant to buyer, `p` contains the buyer's pubkey. 
 - The payment request message can be initiated in two ways, depending on whether the merchant has a server handling payments
 
 ### Message Types
 
 1. Order Creation (buyer → merchant) (subject "order-info")
 ```jsonc
 {
-  "kind": 14,
+  "kind": 15,
   "tags": [
     ["p", "<merchant-pubkey>"],
     ["subject", "order-info"],
     ["order", "<order-id>"],
-    ["item", "<product-id>", "<quantity>"],
-    ["item", "<product-id>", "<quantity>"], // Multiple items possible
-    ["shipping", "<shipping-option-id>"],
+    ["item", "30402:<pubkey>:<d-tag>", "<quantity>"],
+    ["item", "30402:<pubkey>:<d-tag>", "<quantity>"], // Multiple items possible
+    ["shipping", "30406:<pubkey>:<d-tag>"],
+    ["amount", "<total-amount>"],
     ["address", "<shipping-address>"],
```

Comment:
Address could be marked required if the Shipping method requires it, but digital purchases won't need an address, but will require a receiving digital address (npub, email, etc). Proper thinkin'?